xarray_sql/dataframe_protocol/core.py

```diff
@@ -0,0 +1,530 @@
+    """
+    Return the missing value (or "null") representation the column dtype
+    uses, as a tuple ``(kind, value)``.
+
+    Value : if kind is "sentinel value", the actual value. If kind is a bit
+    mask or a byte mask, the value (0 or 1) indicating a missing value. None
+    otherwise.
+    """
+    ...
+
+  @property
+  @abstractmethod
+  def null_count(self) -> t.Optional[int]:
+    """
+    Number of null elements, if known.
+    """
+    ...
+
+  @property
+  @abstractmethod
+  def metadata(self) -> t.Dict[str, t.Any]:
+    """
+    The metadata for the column. See `DataFrame.metadata` for more details.
+    """
+    ...
+
+  @abstractmethod
+  def num_chunks(self) -> int:
+    """
+    Return the number of chunks the column consists of.
+    """
+    ...
+
+  @abstractmethod
+  def get_chunks(
+      self, n_chunks: t.Optional[int] = None
+  ) -> t.Iterable["Column"]:
+    """
+    Return an iterator yielding the chunks.
+
+    See `DataFrame.get_chunks` for details on ``n_chunks``.
+    """
+    ...
+
+  @abstractmethod
+  def get_buffers(self) -> ColumnBuffers:
+    """
+    Return a dictionary containing the underlying buffers.
+
+    The returned dictionary has the following contents:
+
+        - "data": a two-element tuple whose first element is a buffer
+                  containing the data and whose second element is the data
+                  buffer's associated dtype.
+        - "validity": a two-element tuple whose first element is a buffer
+                      containing mask values indicating missing data and
+                      whose second element is the mask value buffer's
+                      associated dtype. None if the null representation is
+                      not a bit or byte mask.
+        - "offsets": a two-element tuple whose first element is a buffer
+                     containing the offset values for variable-size binary
+                     data (e.g., variable-length strings) and whose second
+                     element is the offsets buffer's associated dtype. None
+                     if the data buffer does not have an associated offsets
+                     buffer.
+    """
+    ...
+
+
+class XarrayColumn(Column):
+  """
+  Minimal concrete Column implementation backed by an xarray.DataArray.
+
+  Assumes a 1-D DataArray representing a single logical column. If the
+  underlying array is chunked (e.g., a dask-backed DataArray), this class
+  can expose protocol chunks via num_chunks()/get_chunks().
+
+  Notes:
+    - For lazy (dask) arrays, materializing buffers requires compute. This can
+      be controlled via the allow_compute flag.
+    - Advanced features (categoricals, string offsets, explicit masks) are not
+      implemented here.
+  """
+
+  def __init__(
+      self,
+      dataarray: xr.DataArray,
+      *,
+      allow_compute: bool = True,
+      base_offset: int = 0,
+  ) -> None:
+    if dataarray.ndim != 1:
```

Comment:
Forgive me if I misunderstand; I don't think we want this assert. The trick we exploit in this library is that it is cheap to flatten ndarrays into vectors (a matter of changing metadata/strides). Doing this while handling chunks will take some care, but I believe it is possible.

Reply:
It may be of help to see the "core.py" at the package root to see what I've tried in the past to achieve this effect.